### ISSUE TITLE
audio_recorder: expose afe_fetch_result_t

### ISSUE THREAD

**Issue by stintel** [espressif/esp-adf#1076]
**Is your feature request related to a problem? Please describe.**

For Willow, we're working on a feature to avoid multiple voice assistants in earshot of each other all responding to a voice command. The feature is explained in more detail [here](https://github.com/toverainc/willow/issues/74).

The first step to make this possible has already landed in esp-sr: espressif/esp-sr#79. 

**Describe the solution you'd like**

We would like to be able to access the wake volume from esp-sr in the Willow code.

**Describe alternatives you've considered**

NA

**Additional context**

We managed to make a working PoC, by modifying some ESP-ADF code. These changes can be found [here](https://github.com/toverainc/esp-adf/commits/willow-multiwake). We will gladly open a PR to implement this, but we see our current changes to ESP-ADF as a quick hack, and would like to ask for your input before we open a PR.

**Comment by jason-mao** [comment]
@stintel Yes, the wake volume is useful, we plan to make a break changes to support this feature and improve the compatibility.
